### PR TITLE
WIP: refactor: fix deprecation warnings

### PR DESCRIPTION
**Pull request by erikgb**
GitHub issue resolved: # 

Briefly describe the changes proposed in this PR:

----
PR Author Checklist (see the [contributor guidelines](https://github.com/eclipse/rdf4j/blob/main/CONTRIBUTING.md) for more details):

 - [ ] my pull request is [self-contained](https://rdf4j.org/documentation/developer/merge-strategy/#self-contained-changes-pull-requests-and-commits)
 - [ ] I've added tests for the changes I made
 - [ ] I've applied [code formatting](https://github.com/eclipse/rdf4j/blob/main/CONTRIBUTING.md#code-formatting) (you can use `mvn process-resources` to format from the command line)
 - [ ] I've [squashed](https://rdf4j.org/documentation/developer/squashing) my commits where necessary 
 - [ ] every commit message starts with the issue number (GH-xxxx) followed by a meaningful description of the change

### CODE REVIEW

**Comment by erikgb** [comment]
@jeenbroekstra How do you suggest I split this PR in multiple issues/PRs? I don't think we should merge this as a "big thing". 😉 Fixing deprecation warnings will surely ease https://github.com/eclipse/rdf4j/issues/3559, as this will require bumping versions to a version where the deprecated method is removed.

Is Dependabot enabled on this project at present? Some dependencies seems to lag behind, but there might be other reasons for that.

**Comment by jeenbroekstra** [comment]
> @jeenbroekstra How do you suggest I split this PR in multiple issues/PRs? I don't think we should merge this as a "big thing". 😉 Fixing deprecation warnings will surely ease #3559, as this will require bumping versions to a version where deprecated features are removed.

Perhaps split out by library for which you're fixing deprecations? That way you can also track the reason _why_ you're fixing these more easily: presumably it is because you want to upgrade to a version of that library that no longer supports the deprecated pattern. So I'd suggest to stick in an issue card for each library you want to bump and link a PR that updates its usage.

> Is Dependabot enabled on this project at present? Some dependencies seems to lag behind, but there might be other reasons for that.

It is. But dependabot only flags security concerns in older libraries - it doesn't automatically bump everything for which a new version exists.

File: core/common/io/src/test/java/org/eclipse/rdf4j/common/io/ByteArrayUtilTest.java [hunk 1]
```diff
@@ -72,24 +67,22 @@ public void testFindArray() {
...
 		assertEquals(0, ByteArrayUtil.getInt(new byte[] { 0, 0, 0, 0, 0, 0 }, 0));
 		assertEquals((1 << 24) + (2 << 16) + (3 << 8) + 4, ByteArrayUtil.getInt(new byte[] { 1, 2, 3, 4 }, 0));
 		assertEquals((1 << 24) + (2 << 16) + (3 << 8) + 4, ByteArrayUtil.getInt(new byte[] { 10, 1, 2, 3, 4, 10 }, 1));
 
-		thrown.expect(ArrayIndexOutOfBoundsException.class);
 		ByteArrayUtil.getInt(new byte[] { 1 }, 0);
 	}
 
-	@Test
+	@Test(expected = ArrayIndexOutOfBoundsException.class)
```
Comment (jeenbroekstra) [inline core/common/io/src/test/java/org/eclipse/rdf4j/common/io/ByteArrayUtilTest.java:79]: Fwiw if you're doing JUnit upgrades any way, could we upgrade straight to JUnit 5? That's actually the preferred version now for any new tests - it's just that we have a lot of tests that haven't been upgraded yet... And if you're feeling _really_ generous, also have a look at using AssertJ assertions (though that one is a "preferred optional", not a must :) ).

**Comment by erikgb** [comment]
Thanks, makes a lot of sense. I already started with Spring, and will probably continue with Jackson.

File: core/common/io/src/test/java/org/eclipse/rdf4j/common/io/ByteArrayUtilTest.java (hunk 1 above)
Comment (erikgb) [reply core/common/io/src/test/java/org/eclipse/rdf4j/common/io/ByteArrayUtilTest.java:79]: Good suggestion. I'll take a look. But I'll probably prioritize fixing Jackson before JUnit.

**Comment by erikgb** [comment]
Closing this, as there are currently too many internal deprecation warnings to work effectively with this task. We can return to this when the use of internal deprecated APIs is removed preparing for the next major version.

**Pull request closed without merging**